File: Samples/Digital signatures/AddLtvInfoToSignature/VB.NET/AddLtvInfoToSignature.vb
Imports BitMiracle.Docotic
Imports BitMiracle.Docotic.Pdf

Namespace BitMiracle.Docotic.Pdf.Samples
    Public NotInheritable Class AddLtvInfoToSignature
        Public Shared Sub Main()
            ' NOTE:
            ' Without a license, the library won't allow you to create or read PDF documents.
            ' To get a free time-limited license key, use the form on
            ' https://bitmiracle.com/pdf-library/download

            LicenseManager.AddLicenseData("PUT-LICENSE-HERE")

            Dim outputFileName = "AddLtvInfoToSignature.pdf"
            Using pdf As New PdfDocument("..\Sample Data\signed-cades.pdf")
                pdf.AddLtvInfo()

                Dim timestampOptions = New PdfSignatureTimestampOptions With {
                    .AuthorityUrl = New Uri("http://timestamp.digicert.com")
                }
                Dim incrementalOptions = New PdfSaveOptions() With {
                    .WriteIncrementally = True
                }
                pdf.TimestampAndSave(timestampOptions, outputFileName, incrementalOptions)
            End Using

            Console.WriteLine($"The output is located in {Environment.CurrentDirectory}")

            Process.Start(New ProcessStartInfo(outputFileName) With {.UseShellExecute = True})
        End Sub
    End Class
End Namespace

File: Samples/Digital signatures/AddLtvInfoToSignature/C#/AddLtvInfoToSignature.cs
using System;
using System.Diagnostics;

namespace BitMiracle.Docotic.Pdf.Samples
{
    public static class AddLtvInfoToSignature
    {
        public static void Main()
        {
            // NOTE:
            // Without a license, the library won't allow you to create or read PDF documents.
            // To get a free time-limited license key, use the form on
            // https://bitmiracle.com/pdf-library/download

            LicenseManager.AddLicenseData("PUT-LICENSE-HERE");

            string outputFileName = "AddLtvInfoToSignature.pdf";
            using (var pdf = new PdfDocument(@"..\Sample Data\signed-cades.pdf"))
            {
                pdf.AddLtvInfo();

                var timestampOptions = new PdfSignatureTimestampOptions
                {
                    AuthorityUrl = new Uri("http://timestamp.digicert.com"),
                };
                var incrementalOptions = new PdfSaveOptions()
                {
                    WriteIncrementally = true,
                };
                pdf.TimestampAndSave(timestampOptions, outputFileName, incrementalOptions);
            }

            Console.WriteLine($"The output is located in {Environment.CurrentDirectory}");

            Process.Start(new ProcessStartInfo(outputFileName) { UseShellExecute = true });
        }
    }
}